internal/mesh/internal/controllers/explicitdestinations/controller.go

```diff
@@ -81,31 +86,50 @@ func (r *reconciler) Reconcile(ctx context.Context, rt controller.Runtime, req c
 		return nil
 	}
 
-	// Now get any destinations that we have in the cache that have selectors matching the name
+	// Now get any mapper that we have in the cache that have selectors matching the name
 	// of this CD (name-aligned with workload).
-	destinationIDs := r.destinations.IDsForWorkload(req.ID)
+	destinationIDs := r.mapper.DestinationsForWorkload(req.ID)
+	rt.Logger.Trace("cached destinations IDs", "ids", destinationIDs)
 
 	decodedDestinations, err := r.fetchDestinations(ctx, rt.Client, destinationIDs)
 	if err != nil {
-		rt.Logger.Error("error fetching destinations", "error", err)
+		rt.Logger.Error("error fetching mapper", "error", err)
 		return err
 	}
 
+	if len(decodedDestinations) > 0 {
+		r.mapper.TrackDestinations(req.ID, decodedDestinations)
+	} else {
+		r.mapper.UntrackComputedExplicitDestinations(req.ID)
+	}
+
+	duplicates := findConflicts(decodedDestinations)
```

Comment:
It seems like there are two approaches to this:
1. Only use completely valid destination resources to generate computed explicit destinations.
2. Use valid destinations on destination resources to generate computed explicit destinations.

It looks like the current implementation skips entire destination resources when anything is wrong with them, but I'm concerned that this will cause unexpected behaviors when a single destination on a resource becomes invalid. For example, if a destinations resource specifies `a`, `b` and `c` as destinations and someone deletes service `c`, requests to services `a` and `b` will fail too.

What do you think about this?

Reply:
Yeah another approach would be to partially include destinations without conflicts. The current approach definitely errs a bit more on the conservative side in rejecting the whole resource, but I think this is ok, at least for now, because it would be similar to per-resource validation. Imagine that you instead created a single `Destinations` resource with the same problem (some of the destinations have conflicting addresses, while others don't). In our validation hook, we can validate it, and if there are address/port/path conflicts, return an error and don't save even if there are some destinations that are valid. 

Let me know if this makes sense.

Reply:
I think this is fine for now. We can always circle back to this later when it comes up. 👍